mesonbuild/programs.py

```diff
@@ -287,14 +284,15 @@ def _search_windows_special_cases(self, name: str, command: str) -> T.List[T.Opt
                 return commands
         return [None]
 
-    def _search(self, name: str, search_dir: T.Optional[str]) -> T.List[T.Optional[str]]:
+    def _search(self, name: str, search_dirs: T.List[T.Optional[str]]) -> T.List[T.Optional[str]]:
```

Comment:
I think this should be `Optional[List[str]]`

Reply:
Done in 7eb1997

Reply:
@dcbaker I would disagree here... Looking at the code, it should either be `T.List[T.Optional[str]]` or `T.List[str]`. Passing `None` to this function would result in an error (iterating over `None`).

Though, I would say that `T.List[str]` would make the most sense in this case.

Reply:
regarding the call to `_search` done  in [`ExternalProgram constructor`](https://github.com/mesonbuild/meson/blob/46380b5b1e82a5c2b49299eb35ac185d4fa02e18/mesonbuild/programs.py#L62), I would agree on `T.List[T.Optional[str]]`.

Reply:
I don't think we should be passing `None` here at all, looking at the code. But I guess we can leave it for now.